packages/devtools_app/lib/src/flutter/common_widgets.dart

```diff
@@ -113,3 +116,163 @@ class DefaultTaggedText extends StatelessWidget {
     );
   }
 }
+
+/// A widget that takes two children and lays them out along one axis.
+///
+/// The user can customize the amount of space allocated to each child by
+/// dragging a divider between them.
+class Split extends StatefulWidget {
+  /// Builds a [Split] with [Axis.horizontal] direction.
+  ///
+  /// [firstChild] will be placed before [secondChild] in a [Row].
+  const Split.horizontal({
+    Key key,
+    @required Widget firstChild,
+    @required Widget secondChild,
+    double initialFirstFraction,
+  }) : this._(
+          key,
+          Axis.horizontal,
+          firstChild,
+          secondChild,
+          initialFirstFraction,
+        );
+
+  /// Builds a [Split] with [Axis.vertical] direction.
+  ///
+  /// [firstChild] will be placed before [secondChild] in a [Column].
+  const Split.vertical({
+    Key key,
+    @required Widget firstChild,
+    @required Widget secondChild,
+    double initialFirstFraction,
```

Comment:
you also need to track the minimum size allowed for each split component. that is critical to ensure that users don't accidentally get the UI in a weird state by collapsing one side of the split too much.

Reply:
I've left a TODO to follow up on this.